File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
#include <queue>
using namespace std;

int main(){
	queue<int> q;
	int t;
	cin >> t;
	while(t--){
		string cm; int x;
		cin >> cm;
		if(cm == "push"){
			cin >> x;
			q.push(x);
		}
		else if(cm == "pop"){
			cout << (q.empty()?-1:q.front()) << endl;
			if(q.empty() == false) q.pop();
		}
		else if(cm == "size"){
			cout <<(q.empty()?0:q.size())<< endl;}
		else if(cm == "empty"){cout << (q.empty()?1:0) << endl;}
		else if(cm == "front"){cout << (q.empty()?-1:q.front())<<endl;}
		else if(cm == "back"){cout << (q.empty()?-1:q.back())<<endl;}
	}
	return 0;
}

File: 백준/Silver/10845. 큐/큐.py
import sys
input = sys.stdin.readline

n = int(input())
q = []
for _ in range(n):
    a = input().split()
    if a[0] == "push":
        q.insert(0, int(a[1]))
    elif a[0] == "front":
        if q: print(q[-1])
        else: print(-1)
    elif a[0] == "back":
        if q: print(q[0])
        else: print(-1)
    elif a[0] == "pop":
        if q: 
            print(q[-1])
            q.pop()
        else: print(-1)
    elif a[0] == "size":
        print(len(q))
    else:
        if q: print(0)
        else: print(1)